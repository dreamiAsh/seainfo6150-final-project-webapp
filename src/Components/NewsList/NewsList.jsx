import React, {useEffect, useState} from "react";
import styles from "./NewsList.module.css";
import { Link } from "react-router-dom";

const NewsList = (props) => {

  const [fetchedData2, setFetchedData2] = useState({});
  useEffect(() => {
    const fetchData2 = async () => {
      // put data fetching code here!
      // performs a GET request
      const response = await fetch(
        "http://demo9533681.mockable.io/whatList"
      );
      const responseJson = await response.json();
      setFetchedData2(responseJson);
    };
    fetchData2();
  }, [fetchedData2]);

  const [fetchedData, setFetchedData] = useState({});
  useEffect(() => {
    const fetchData = async () => {
      // put data fetching code here!
      // performs a GET request
      const response = await fetch(
        "http://demo9533681.mockable.io/popularList"
      );
      const responseJson = await response.json();
      setFetchedData(responseJson);
    };
    fetchData();
  }, [fetchedData]);

  const myList = Object.values(fetchedData)
  const mappedList = myList.map((listItem) => (
    <Link forceRefresh={true} className={styles.item} to={`/news/${listItem.title}/${listItem.date}/${listItem.imageId}/${listItem.content}`}>{listItem.title}</Link>
  ))

  const myList2 = Object.values(fetchedData2)
  const mappedList2 = myList2.map((listItem) => (
    <Link forceRefresh={true} className={styles.item} to={`/news/${listItem.title}/${listItem.date}/${listItem.imageId}/${listItem.content}`}>{listItem.title}</Link>
  ))

  return (
    <div className={styles.box}>
        <div className={styles.leftBox}>
            <h4 className={styles.itemTitle}>What's new</h4>
            <ul>{mappedList}</ul>
        </div>
        <div className={styles.rightBox}>
            <h4 className={styles.itemTitle}>Most popular</h4>
            <ul>{mappedList2}</ul>
        </div>
    </div>
  );
};

export default NewsList;
