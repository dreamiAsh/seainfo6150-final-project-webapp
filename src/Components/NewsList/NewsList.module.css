.box{
    width: 100%;
    height: 200px;
    padding-top: 45px;
    display: grid;
    flex-flow: column-reverse;
    justify-content: space-around;
    align-items: stretch;
    grid-template-columns: repeat(2, 30fr);
}
.itemTitle{
    text-align: center;
    margin-bottom: 8px;
    font-family: Courier New;
    font-size: 30px;
}
.item{
    text-align: center;
    margin-bottom: 5px;
    color: black;
    text-decoration: none;
    display: block;
}
.item:hover{
    color: orange;
}