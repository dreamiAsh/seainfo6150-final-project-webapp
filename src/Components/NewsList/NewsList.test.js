import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewsList from "./NewsList.jsx";

it("renders the footer component correctly", () => {
  const { container } = render(<NewsList />);
  expect(container).toMatchSnapshot();
});