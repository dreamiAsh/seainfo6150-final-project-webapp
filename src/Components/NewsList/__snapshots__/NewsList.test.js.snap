// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the footer component correctly 1`] = `
<div>
  <div
    class="box"
  >
    <div
      class="leftBox"
    >
      <h4
        class="itemTitle"
      >
        What's new
      </h4>
      <ul />
    </div>
    <div
      class="rightBox"
    >
      <h4
        class="itemTitle"
      >
        Most popular
      </h4>
      <ul />
    </div>
  </div>
</div>
`;
